src/commands/project/deploy/quick.ts

```diff
@@ -84,9 +84,24 @@ export default class DeployMetadataQuick extends SfCommand<DeployResultJson> {
     const [{ flags }, cache] = await Promise.all([this.parse(DeployMetadataQuick), DeployCache.create()]);
 
     // This is the ID of the validation request
-    const jobId = cache.resolveLatest(flags['use-most-recent'], flags['job-id'], false);
+    let jobId: string;
```

Comment:
I'm not a fan of the `{} as` stuff.  I think I'd rather see a clean function that encapsulates all the complexity and gets enough inputs to

1. check the cache
2.  deal with target-org
3.  return `[org: Org, jobId:string]` or something like that.

5.   DeployOptions is really only used to get to a target-org so creating empty ones seems unnecessary.